src/main/java/hudson/plugins/promoted_builds/Status.java

```diff
@@ -303,7 +303,8 @@ public boolean isManuallyApproved(){
      * Schedules a new build.
      */
     public void doBuild(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException {
-        if(!getTarget().hasPermission(Promotion.PROMOTE))
+        // JENKINS-24271 Re-execute and Force permissions must be separated
+        if(!getTarget().hasPermission(Promotion.REEXECUTE))
             return;
```

Comment:
It's an issue in the original code, but I would expect a "permission denied" error here